Sources/SourceKit/sourcekitd/SwiftLanguageServer.swift

```diff
@@ -547,6 +548,81 @@ extension SwiftLanguageServer {
     // FIXME: cancellation
     _ = handle
   }
+
+  func foldingRange(_ req: Request<FoldingRangeRequest>) {
+
+    guard let snapshot = documentManager.latestSnapshot(req.params.textDocument.url) else {
+      log("failed to find snapshot for url \(req.params.textDocument.url)")
+      req.reply(nil)
+      return
+    }
+
+    let skreq = SKRequestDictionary(sourcekitd: sourcekitd)
+    skreq[keys.request] = requests.editor_open
+    skreq[keys.name] = "IBSwiftAnnotationProvider:" + snapshot.document.url.path
```

Comment:
Hahaha I left it in because I was trying to cover the day Xcode implements the LSP. I assumed it was important for this request to continue sending it. Is just the URL enough or should we come up with another prefix?

Reply:
Use another prefix, otherwise it will conflict with the normal `textDocument/didOpen` code path.  Something like "FoldingRanges:" would be fine.